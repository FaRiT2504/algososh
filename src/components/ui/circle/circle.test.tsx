import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

const renderer = require('react-test-renderer');

describe("Тест компонента Circle", () => {

  it('Тест компонента Circle без буквы', () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с буквами', () => {
    const tree = renderer.create(<Circle letter='A' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с head', () => {
    const tree = renderer.create(<Circle head='head' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с react-элементом в head', () => {
    const tree = renderer
      .create(<Circle head={<Circle />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с tail', () => {
    const tree = renderer.create(<Circle tail='tail' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с react-элементом в tail', () => {
    const tree = renderer
      .create(<Circle tail={<Circle />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Тест компонента Circle с index', () => {
    const tree = renderer.create(<Circle index={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle с пропcом isSmall === true', () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle в состоянии default', () => {
    const tree = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle в состоянии changing', () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Тест компонента Circle в состоянии modified', () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

})