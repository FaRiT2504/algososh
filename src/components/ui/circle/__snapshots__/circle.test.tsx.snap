// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Circle Тест компонента Circle без буквы 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle в состоянии changing 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle в состоянии default 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle в состоянии modified 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с head 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с index 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с react-элементом в head 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
      data-cy="circleWrapper"
      data-testid="circle-test"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с react-элементом в tail 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  >
    <div
      className="content "
      data-cy="circleWrapper"
      data-testid="circle-test"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с tail 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circleTail"
  >
    tail
  </div>
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с буквами 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Тест компонента Circle Тест компонента Circle с пропcом isSmall === true 1`] = `
<div
  className="content "
  data-cy="circleWrapper"
  data-testid="circle-test"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;
